src/electronic-ids/ms-cryptoapi/MsCryptoApiElectronicID.cpp

```diff
@@ -21,49 +21,60 @@
  */
 
 #include "MsCryptoApiElectronicID.hpp"
-#include "../scope.hpp"
-
-#include <openssl/x509v3.h>
-#include <openssl/err.h>
+#include "../x509.hpp"
 
 using namespace pcsc_cpp;
 
-namespace
+namespace electronic_id
 {
 
-using namespace electronic_id;
+JsonWebSignatureAlgorithm MsCryptoApiElectronicID::authSignatureAlgorithm() const
+{
+    // TODO: PS256
+    return getAuthAlgorithmFromCert(certData);
+}
 
-JsonWebSignatureAlgorithm getESAlgorithmFromCert(const byte_vector& cert)
+byte_vector MsCryptoApiElectronicID::signWithAuthKey(const byte_vector& /* pin */,
+                                                     const byte_vector& hash) const
 {
-    const unsigned char* certPtr = cert.data();
-    auto x509 = SCOPE_GUARD(X509, d2i_X509(nullptr, &certPtr, long(cert.size())));
-    if (!x509) {
-        THROW(MsCryptoApiError, "Failed to create X509 object from certificate");
+    if (certType != CertificateType::AUTHENTICATION) {
+        THROW(WrongCertificateTypeError,
+              "This electronic ID does not support sigining with the authentication key. "
```

Comment:
Typo: `sigining`

Here and bunch of other locations, probably worth grepping for it in whole repo.

Reply:
Thanks